src/org/omegat/core/team2/impl/SVNRemoteRepository2.java

```diff
@@ -120,8 +119,9 @@ public void switchToVersion(String version) throws Exception {
             ourClientManager.getUpdateClient().doCheckout(url, baseDirectory, SVNRevision.HEAD,
                     toRev, SVNDepth.INFINITY, false);
             Log.logInfoRB("SVN_FINISH", "checkout");
-        } catch (Exception ex) {
+        } catch (SVNException ex) {
```

Comment:
Now we don't handle any non-`SVNException`s. Can you add a clause for `Exception` that just logs and re-throws it, like in `commit()`?

Reply:
This try block throws only `SVNException`, so I just narrowed exception scope. I can revert this change but can't see why.

Reply:
The only *declared* exception is `SVNException`. Anything else that might happen due to e.g. a bug in SVNKit will not be caught.

Also, you might take a look at the SVNKit code and say "they are wrapping all exceptions with SVNException, so it's OK", and you might be right for this version of SVNKit. But you don't know about future versions.

So the safe play is to catch everything.

Reply:
Thanks for explanation. I just reverted the change because extra catch won't bring any functionality here.

Reply:
That's good, too. Thanks.